### DISCUSSION TITLE
Nametags over head?

### DISCUSSION THREAD

**Discussion by thebonbon** [EveronLife/EveronLife#151] [Polls]
Players currently have their name shown above their head.

Should this feature be enabled or disabled by default?

**Comment by thebonbon** [comment]
I think it should be disabled by default since it breaks the whole RP immersion.

**Reply by Arkensor** [reply]
Do you prefer people introducing their name via voice chat and or showing their id cards to the police etc?

**Reply by thebonbon** [reply]
Yes this. Why even ask for a name if you just see it. 
What if you are hiding in a bush and the nametag just floats above it. Its just not for a RP server imo.

I would even delete this feature if possible (But its a Framework and should up to the server, but we can disable it by default).

**Reply by BenRauzi** [reply]
Honestly I don't mind on this one. With A3 it was most common to see servers using name tags so that'd probably be what most would prefer once again. 

But regardless I think by default there should definitely be _some_ identifier (e.g. a shortened player ID, some good feedback I had in things I developed was IDs that could be customised to add names to players if you want to and this became pretty common in the last few years too) that can be used for reporting (staff would be an nightmare without unique identifiers) and also identifying other players you've interacted with (e.g. so you don't arrest the wrong person just wearing the same clothes)

That is however extra work.

**Reply by BenRauzi** [reply]
@thebonbon 
> What if you are hiding in a bush and the nametag just floats above it.

 Just to note this should be preventable. I'm not super familiar with the scripting yet but this was possible in a3 even on bushes without collision, I assume they're supporting it in some way here.

**Reply by thebonbon** [reply]
>  (e.g. so you don't arrest the wrong person just wearing the same clothes)
>
Yes you dont know if its the right person. Thats the fun of good RP.

Its pretty common cause all the RP was just PVP and you had to report alot of players.
Servers should fix the base problem first to not need IDs or names to report people.

I dont know if im too optimistic to try and fix the all the RP with the Framework, but im trying with every little feature ^^

And a better solution for staff is a good logging system (Shots fired, dmg, deaths etc..)

**Comment by thebonbon** [comment]
(Hardcore) Servers thats focus on good RP will disable it anyway.
So why not disable it by default and if a Server wants / needs it for reporting players or anything else they can activate it.

With this and many other features we can hopefully create a new and better "meta" of RP servers that dont need IDs or Names to report players.

**Comment by CheekyLad** [comment]
I would say default on so you can report players if they do trouble things with their name in case they don't tell or show you

**Reply by CheekyLad** [reply]
If a server owner decides he don't want that he can still disable it

**Reply by thebonbon** [reply]
Why not the other way around?

**Comment by Arkensor** [comment]
From a technical standpoint, since I just now had to dig into the nametag system, I'll add config settings to selectively show nametags for players, or only NPCs or you disable them altogether. Later we can also make it so that you only see names of your "group" or maybe only nametags of people you "met" who introduced them to you. So maybe there is a social interaction to "befriend" somebody that will do things like add them to your phonebook and also show their nametags. People you did not "befriend" have neither.
